packages/ui-react-utils/src/callRenderProp.ts

```diff
@@ -43,21 +34,7 @@ import React, {
  * @param value
  * @param props
  */
-function callRenderProp<P>(
-  value:
-    | keyof ReactHTML
-    | keyof ReactSVG
-    | FunctionComponent<P>
-    | ClassType<
-        P,
-        ClassicComponent<P, ComponentState>,
-        ClassicComponentClass<P>
-      >
-    | ComponentClass<P>
-    | ReactNode
-    | (() => ReactNode),
-  props: P = {} as P
-) {
+function callRenderProp<P>(value: Renderable<P>, props: P = {} as P) {
```

Comment:
Extracting this type resulted in lots of small changes. This was needed because React 18 has better checks and we could broaden the type in cases where a prop value ended up as a parameter for this function.